test/jasmine/tests/pie_test.js

```diff
@@ -2064,3 +2064,72 @@ describe('pie uniformtext', function() {
         .then(done);
     });
 });
+
+describe('pie value format', function() {
+    'use strict';
+
+    var gd;
+
+    beforeEach(function() {
+        gd = createGraphDiv();
+    });
+
+    afterEach(destroyGraphDiv);
+
+    it('should handle rounding big & small numbers', function(done) {
+        Plotly.newPlot(gd, [{
+            type: 'pie',
+            textinfo: 'value',
+            values: [
+                123456789012,
+                12345678901.2,
+                1234567890.12,
+                123456789.012,
+                12345678.9012,
+                1234567.89012,
+                123456.789012,
+                12345.6789012,
+                1234.56789012,
+                123.456789012,
+                12.3456789012,
+                1.23456789012,
+                0.123456789012,
+                0.0123456789012,
+                0.00123456789012,
+                0.000123456789012,
+                0.0000123456789012,
+                0.00000123456789012,
```

Comment:
heh you just needed to add one more zero to get `e-` 😅 But I think it makes sense to keep `e+/-` in the result when that's what `toPrecision` does - eventually when the number gets big enough (or small enough) that'll be necessary instead of `round`.

Reply:
Good call. Revised in 5891651.